### DISCUSSION TITLE
RDF calculation

### DISCUSSION THREAD

**Discussion by AnuragKr** [deepmodeling/deepmd-kit#1861] [Q&A] [answered]
Hello,
I am able to create RDF from deepmd-kit interatomic potential by following this tutorial [link](https://github.com/CSIprinceton/workshop-july-2022/tree/main/hands-on-sessions/day-1/2-molecular-dynamics-deepmd/liquid_water/300K) but for comparison, I want to create with DFT also. How should I go about it?

The goal is to reproduce this graph--
![rdf_comp](https://user-images.githubusercontent.com/10074710/184847871-dfc68280-f900-4498-a1ac-e0457863d867.png)

Dataset I am using examples/water/se_e2_a 

I think I need to change only the argument of pair_style. But I don't know what should I put. Please help in resolving this issue

LAMMPS Code ----
#########
units metal
atom_style atomic
boundary p p p
read_data water.lmp
pair_style deepmd ../../frozen_model_compressed.pb
pair_coeff * *

timestep 0.0005

group Oatoms type 1
group Hatoms type 2

fix 1 all npt temp 300 300 0.05 iso 1.0 1.0 0.5

thermo_style custom step temp ke pe etotal press vol
thermo 100
thermo_modify format line "%d %.6e %.6e %.6e %.6e %.6e %.6e %.6e %.6e %.6e"

dump my_dump all custom 100 h2o.xyz id type x y z vx vy vz
dump_modify my_dump sort id
dump my_dump2 all atom 100 water.lammps-dump-text

compute myRDF all rdf 100 1 1 1 2 2 2 
fix 2 all ave/time 100 1 100 c_myRDF[*] file h2o.rdf mode vector ave running

compute msd_O Oatoms msd
compute msd_H Hatoms msd
fix store_msd_O all vector 10 c_msd_O[4]
fix store_msd_H all vector 10 c_msd_H[4]
variable fitslope_O equal slope(f_store_msd_O)/6/(10*dt)
variable fitslope_H equal slope(f_store_msd_H)/6/(10*dt)
fix 3 all ave/time 100 1 100 c_msd_O[4] c_msd_H[4] v_fitslope_O v_fitslope_H file diffusion.txt

run 100000
##########

**Comment by Zhang-pchao** [comment]
You can refer to the existing codes about calculating RDF, such as VMD and MDAnalysis.

**Reply by AnuragKr** [reply]
Can you please explain like how to calculate rdf with binary data given box.npy, coord.npy, force.npy, and energy.npy.

**Reply by Zhang-pchao** [reply]
Sorry, I don't know. RDF can be calculated from MD/DPMD/AIMD trajectories.

**Reply by AnuragKr** [reply]
@njzjz   1. How can I generate RDF from binary data given in .npy format? 
              2. Which AIMD software is being used for data generation and simulation?

**Reply by Zhang-pchao** [reply]
RDF can be calculated based on atomic types (type_map.raw and type.raw) and coordinates (coord.npy or coord.raw), but the error is large due to less statistical information. Why not choose to process RDF from the MD trajectory?

**Comment by AnuragKr** [comment, marked as answer]
We can create RDF directly using deepmd binary data using following code.--
Link - https://github.com/deepmodeling/dpdata/blob/master/dpdata/md/rdf.py
